src/Windows/Avalonia.Win32/WinRT/Composition/WinUICompositorConnection.cs

```diff
@@ -124,15 +128,25 @@ private void RunLoop()
             var cts = new CancellationTokenSource();
             AppDomain.CurrentDomain.ProcessExit += (sender, args) =>
                 cts.Cancel();
-                
+            var asyncActionCompletedHandler = new RunLoopHandler(this);
             using (var act = _compositor5.RequestCommitAsync())
-                act.SetCompleted(new RunLoopHandler(this));
+            {
+                act.SetCompleted(asyncActionCompletedHandler);
+            }
+            _semaphore.Wait();
```

Comment:
I believe that code with `ManualResetEvent` would be less complicated

Reply:
Thanks